engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -164,15 +271,23 @@ Object doWarning(
       }
     }
 
-    arguments[thisArgumentPosition] = self.getValue();
-    ArrayRope<Warning> warnings = self.getReassignedWarnings(this);
```

Comment:
So we are getting rid of the reassigned warnings altogether? Are we still going to be able to find out the origin of the warning?

Reply:
`getWarnings` does the reassignment AFAIK https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/java/org/enso/interpreter/runtime/error/WithWarnings.java#L118